dts/riscv/hpmicro/hpm6750.dtsi

```diff
@@ -0,0 +1,178 @@
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				compatible = "riscv,cpu-intc";
+			};
+		};
+		cpu1: cpu@1 {
+			compatible = "riscv";
+			device_type = "cpu";
+			reg = <1>;
+			status = "okay";
+			riscv,isa = "rv32imafdcxandes";
+			i-cache-line-size = <32>;
+			d-cache-line-size = <32>;
+			CPU1_intc: interrupt-controller {
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				compatible = "riscv,cpu-intc";
+			};
+		};
+
+		fgpio: peripheral@C0000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0xC0000 0x2000>;
+		};
+	};
+
+	soc {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "hpmicro,hpm6750";
+		ranges;
+		clk: clock-controller@f4100000 {
+			compatible = "hpmicro,hpm-clock";
+			reg = <0xf4100000 0x100>;
+			clock-frequency = <816000000>;
+			reg-names = "clk";
+			status = "okay";
+			clock-sys-core = <HPM_CLOCK_CPU0>;
+			ram-up-time = <(32*1000*9)>;
+			sysctl-present = "sysctl-preset-0";
+			#clock-cells = <3>;
+		};
+
+		mtimer: timer@e6000000 {
+			compatible = "andestech,machine-timer";
+			reg = <0xe6000000 0x10>;
+			interrupts-extended = <&CPU0_intc 7 &CPU1_intc 7>;
+		};
+
+		flash0: flash@80000000 {
+			compatible = "soc-nv-flash";
+			reg = <0x80000000 DT_SIZE_M(16)>;
+		};
+
+		ilm: memory@0 {
+			compatible = "zephyr,memory-region";
+			reg = <0x0 DT_SIZE_K(256)>;
+			zephyr,memory-region = "ITCM";
+		};
+
+		dlm: memory@80000 {
+			compatible = "zephyr,memory-region";
+			reg = <0x080000 DT_SIZE_K(256)>;
+			zephyr,memory-region = "DTCM";
+		};
+
+		sram: memory@1080000 {
+			device_type = "memory";
+			compatible = "mmio-sram";
+			reg = <0x01080000 DT_SIZE_K(1024)>;
+		};
+
+		core0_lm_slv: memory@1000000 {
+			device_type = "memory";
+			reg = <0x1000000 DT_SIZE_K(512)>;
+		};
+
+		core1_lm_slv: memory@1180000 {
+			device_type = "memory";
+			reg = <0x1180000 DT_SIZE_K(256)>;
+		};
+
+		noncache: memory@11c0000 {
+			device_type = "memory";
+			reg = <0x11c0000 DT_SIZE_K(256)>;
+		};
+
+		plic0: interrupt-controller@e4000000 {
+			compatible = "sifive,plic-1.0.0";
+			#address-cells = <1>;
+			#interrupt-cells = <2>;
+			interrupt-controller;
+			reg = <	0xe4000000 0x00001000
+				0xe4002000 0x00000800
+				0xe4200000 0x00010000 >;
+			reg-names = "prio", "irq_en", "reg";
+			riscv,max-priority = <255>;
+			riscv,ndev = <1023>;
+			interrupts-extended = <&CPU0_intc 0 &CPU1_intc 0>;
+		};
+
+		plic_sw0: interrupt-controller@e6400000 {
+			compatible = "andestech,plic_sw";
+			#address-cells = <1>;
+			#interrupt-cells = <2>;
+			interrupt-controller;
+			reg = <0xe6400000 0x00400000>;
+			riscv,max-priority = <255>;
+			riscv,ndev = <1023>;
+			interrupts-extended = <&CPU0_intc 3 &CPU1_intc 3>;
+		};
+
+		gpio0: peripheral@f0000000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0xf0000000 0x4000>;
+		};
+
+		gpio1: peripheral@f0004000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0xf0004000 0x4000>;
+		};
```

Comment:
compatible? missing `gpio-controller`? node name doesn't follow dt recommended names either.

Reply:
From the chip point of view, `gpio0` and `gpio1` are the real `gpio controllers`, `gpioa` `gpiob` ... `gpioz` is a grouping of them, if you use `gpio0` `pin100` directly this way to control io, the effect is very poor. The grouping is implemented in `hpm6750evkmini.dts`, `gpiob` is in the `gpio0` controller.